packages/server/src/fhir/lookups/address.ts

```diff
@@ -83,43 +82,40 @@ export class AddressTable extends LookupTable<Address> {
    * Attempts to reuse existing Addresses if they are correct.
    * @param client - The database client.
    * @param resource - The resource to index.
+   * @param create - True if the resource should be created (vs updated).
    * @returns Promise on completion.
    */
-  async indexResource(client: PoolClient, resource: Resource): Promise<void> {
+  async indexResource(client: PoolClient, resource: Resource, create: boolean): Promise<void> {
+    if (!create) {
+      await this.deleteValuesForResource(client, resource);
+    }
+
     const addresses = this.getIncomingAddresses(resource);
     if (!addresses || !Array.isArray(addresses)) {
       return;
     }
 
     const resourceType = resource.resourceType;
     const resourceId = resource.id as string;
-    const existing = await this.getExistingValues(client, resourceType, resourceId);
-
-    if (!compareArrays(addresses, existing)) {
-      if (existing.length > 0) {
-        await this.deleteValuesForResource(client, resource);
-      }
+    const values = [];
 
-      const values = [];
-
-      for (let i = 0; i < addresses.length; i++) {
-        const address = addresses[i];
-        values.push({
-          id: randomUUID(),
-          resourceId,
-          index: i,
-          content: stringify(address),
-          address: formatAddress(address),
-          city: address.city?.trim(),
-          country: address.country?.trim(),
-          postalCode: address.postalCode?.trim(),
-          state: address.state?.trim(),
-          use: address.use?.trim(),
-        });
-      }
-
-      await this.insertValuesForResource(client, resourceType, values);
+    for (let i = 0; i < addresses.length; i++) {
+      const address = addresses[i];
+      values.push({
+        id: randomUUID(),
+        resourceId,
+        index: i,
+        content: stringify(address),
+        address: formatAddress(address),
+        city: address.city?.trim(),
+        country: address.country?.trim(),
+        postalCode: address.postalCode?.trim(),
+        state: address.state?.trim(),
+        use: address.use?.trim(),
+      });
```

Comment:
This looks like a case where a straightforward `map()` more clearly expresses the intent of the code:

```suggestion
const values = addresses.map((address, index) => ({
  id: randomUUID(),
  resourceId,
  index,
  content: stringify(address),
  address: formatAddress(address),
  city: address.city?.trim(),
  country: address.country?.trim(),
  postalCode: address.postalCode?.trim(),
  state: address.state?.trim(),
  use: address.use?.trim(),
}));
```